spruned/app.py

```diff
@@ -157,10 +157,13 @@ def start():  # pragma: no cover
             from spruned.application.logging_factory import Logger
             from spruned.application.database import sqlite
             from spruned.builder import repository
+
             migrations.run(sqlite)
 
             version = repository.blockchain.get_db_version()
-            if version != repository.blockchain.current_version:
+            if version != repository.blockchain.current_version and version != None:
```

Comment:
I don't understand this change. I'm afraid this way the erase wouldn't get triggered on new updates?

Reply:
I think this issue is more annoying than it seems. 
The erase method:

- closes the leveldb connection
- delete the storage file
- reopen a new leveldb connection

However, some clients continue using the previous connection, failing.
This is due to a lack of a decent dependency injection framework, that is something is not worth the effort at the moment to implement (this may change in the future with DDD I'm going to approach in the next releases, but not now).
There are three occurrences, if I remember well, of the leveldb client at the moment. The blockchain repository, the cache agent and the transactional decorator.

We have to ensure that the erase task is made when needed, and all the occurrences to the new client are referred.